diff --git a/src/platforms/rcore_desktop_sdl.c b/src/platforms/rcore_desktop_sdl.c
index a201f2c..3d0e4a1 100644
--- a/src/platforms/rcore_desktop_sdl.c
+++ b/src/platforms/rcore_desktop_sdl.c
@@ -1342,10 +1342,17 @@ void PollInputEvents(void)
     }
 
     // Register previous touch states
-    for (int i = 0; i < MAX_TOUCH_POINTS; i++) CORE.Input.Touch.previousTouchState[i] = CORE.Input.Touch.currentTouchState[i];
+    for (int i = 0; i < MAX_TOUCH_POINTS; i++)
+    {
+      CORE.Input.Touch.previousTouchState[i] = CORE.Input.Touch.currentTouchState[i];
+      // todo clear touch position?
+      // CORE.Input.Touch.position[i].x = -1;
+      // CORE.Input.Touch.position[i].y = -1;
+      // CORE.Input.Touch.pointId[i] = -1;
+    }
 
-    // Map touch position to mouse position for convenience
-    CORE.Input.Touch.position[0] = CORE.Input.Mouse.currentPosition;
+    // // Map touch position to mouse position for convenience
+    // CORE.Input.Touch.position[0] = CORE.Input.Mouse.currentPosition;
 
     int touchAction = -1;       // 0-TOUCH_ACTION_UP, 1-TOUCH_ACTION_DOWN, 2-TOUCH_ACTION_MOVE
     bool realTouch = false;     // Flag to differentiate real touch gestures from mouse ones
@@ -1583,13 +1590,19 @@ void PollInputEvents(void)
             } break;
             case SDL_FINGERUP:
             {
+                int count = CORE.Input.Touch.pointCount;
                 UpdateTouchPointsSDL(event.tfinger);
+                CORE.Input.Touch.pointCount = count;
+
                 touchAction = 0;
                 realTouch = true;
             } break;
             case SDL_FINGERMOTION:
             {
+                int count = CORE.Input.Touch.pointCount;
                 UpdateTouchPointsSDL(event.tfinger);
+                CORE.Input.Touch.pointCount = count;
+
                 touchAction = 2;
                 realTouch = true;
             } break;
@@ -1738,28 +1751,42 @@ void PollInputEvents(void)
         {
             // Process mouse events as touches to be able to use mouse-gestures
             GestureEvent gestureEvent = { 0 };
-
             // Register touch actions
             gestureEvent.touchAction = touchAction;
 
-            // Assign a pointer ID
-            gestureEvent.pointId[0] = 0;
-
-            // Register touch points count
-            gestureEvent.pointCount = 1;
-
-            // Register touch points position, only one point registered
-            if (touchAction == 2 || realTouch) gestureEvent.position[0] = CORE.Input.Touch.position[0];
-            else gestureEvent.position[0] = GetMousePosition();
-
-            // Normalize gestureEvent.position[0] for CORE.Window.screen.width and CORE.Window.screen.height
-            gestureEvent.position[0].x /= (float)GetScreenWidth();
-            gestureEvent.position[0].y /= (float)GetScreenHeight();
+            if (realTouch)
+            {
+              // Register touch points count
+              gestureEvent.pointCount = CORE.Input.Touch.pointCount;
+
+              // we want to track every touch.
+              for (int i = 0; i < CORE.Input.Touch.pointCount; i++)
+              {
+                gestureEvent.pointId[i] = i;
+                gestureEvent.position[i].x = CORE.Input.Touch.position[i].x / (float)GetScreenWidth();
+                gestureEvent.position[i].y = CORE.Input.Touch.position[i].y / (float)GetScreenWidth();
+              }
+            }
+            else
+            {
+              // Register touch points count
+              gestureEvent.pointCount = 1;
+              // Assign a pointer ID
+              gestureEvent.pointId[0] = 0;
+              // Register touch points position, only one point registered
+              if (touchAction == 2 || realTouch) gestureEvent.position[0] = CORE.Input.Touch.position[0];
+              else gestureEvent.position[0] = GetMousePosition();
+
+              // Normalize gestureEvent.position[0] for CORE.Window.screen.width and CORE.Window.screen.height
+              gestureEvent.position[0].x /= (float)GetScreenWidth();
+              gestureEvent.position[0].y /= (float)GetScreenHeight();
+            }
 
             // Gesture data is sent to gestures-system for processing
             ProcessGestureEvent(gestureEvent);
 
             touchAction = -1;
+            realTouch = false;
         }
 #endif
     }
