diff --git a/src/mbedtls/engine.c b/src/mbedtls/engine.c
index 3d9e2f9..6dc6dad 100644
--- a/src/mbedtls/engine.c
+++ b/src/mbedtls/engine.c
@@ -91,7 +91,7 @@ struct mbedtls_engine {
     mbedtls_ssl_session *session;
 
     io_ctx io;
-    uv_os_fd_t io_fd;
+    uv_os_sock_t io_fd;
     io_read read_f;
     io_write write_f;
 
@@ -111,7 +111,7 @@ static int mbedtls_set_own_cert(tls_context *ctx, tlsuv_private_key_t key, tlsuv
 tlsuv_engine_t new_mbedtls_engine(tls_context *ctx, const char *host);
 
 static void mbedtls_set_io(tlsuv_engine_t, io_ctx , io_read , io_write );
-static void mbedtls_set_fd(tlsuv_engine_t, uv_os_fd_t );
+static void mbedtls_set_fd(tlsuv_engine_t, uv_os_sock_t );
 
 static tls_handshake_state mbedtls_hs_state(tlsuv_engine_t engine);
 static tls_handshake_state
@@ -720,7 +720,7 @@ static void mbedtls_set_io(tlsuv_engine_t e, io_ctx io, io_read read_f, io_write
     mbedtls_ssl_set_bio(eng->ssl, eng, engine_io_write, engine_io_read, NULL);
 }
 
-static void mbedtls_set_fd(tlsuv_engine_t e, uv_os_fd_t fd) {
+static void mbedtls_set_fd(tlsuv_engine_t e, uv_os_sock_t fd) {
     struct mbedtls_engine *eng = (struct mbedtls_engine *) e;
     assert(eng->io == NULL);
     eng->io_fd = fd;
